docs/source/library-user-guide/adding-udfs.md

```diff
@@ -74,11 +139,10 @@ assert_eq!(result, &Int64Array::from(vec![Some(2), None, Some(4)]));
 
 The challenge however is that DataFusion doesn't know about this function. We need to register it with DataFusion so that it can be used in the context of a query.
 
-### Registering a Scalar UDF
+#### Registering a Scalar UDF
 
 To register a Scalar UDF, you need to wrap the function implementation in a [`ScalarUDF`] struct and then register it with the `SessionContext`.
 DataFusion provides the [`create_udf`] and helper functions to make this easier.
-There is a lower level API with more functionality but is more complex, that is documented in [`advanced_udf.rs`].
```

Comment:
I doubt a little that whether my PR is necessary, it seems that there's already a updated link that helps user create scalarUDF by `impl trait`. 🤔 What do you think ? @alamb

Reply:
I think the value of this library guide is the surrounding text, so it seems ok to me to have the code replicated in two places